toolchain/sem_ir/formatter.cpp

```diff
@@ -139,6 +139,11 @@ class Formatter {
     out_ << "\nclass ";
     FormatClassName(id);
 
+    if (class_info.generic_id.is_valid()) {
+      out_ << "\n    ";
+      FormatGeneric(class_info.generic_id);
+    }
```

Comment:
Had you considered moving the `is_valid` + indentation into `FormatGeneric` (or a different shared call), to make it consistent between all calls? I see 3 places doing this in the change.

Reply:
Yeah, I'd considered it. I'm on the fence -- it doesn't really seem like it should be `FormatGeneric`'s responsibility to know how the enclosing context is formatted, but it is otherwise being duplicated. Hm, I think I can see a way to handle this a bit more generically...

Reply:
Done.